From 9e70f86884e2dda6773a3d88bd1cd786ebbfa8b7 Mon Sep 17 00:00:00 2001
From: Michael Biebl <biebl@debian.org>
Date: Mon, 27 Jun 2016 22:03:01 +0200
Subject: [PATCH] avahi-autoipd: fix dhclient hooks to check for avahi-autoipd
 before trying to run it

In Debian the package can be in uninstalled but not purged state, i.e.
the dhclient hooks are present but the avahi-autoipd binary is not.
---
 avahi-autoipd/dhclient-enter-hook.in               |  2 +-
 avahi-autoipd/dhclient-exit-hook.in                |  2 +-
 ...pd-fix-dhclient-hooks-to-check-for-avahi-.patch | 41 ++++++++++++++++++++++
 debian/patches/series                              |  1 +
 4 files changed, 44 insertions(+), 2 deletions(-)
 create mode 100644 debian/patches/0001-avahi-autoipd-fix-dhclient-hooks-to-check-for-avahi-.patch
 create mode 100644 debian/patches/series

diff --git a/avahi-autoipd/dhclient-enter-hook.in b/avahi-autoipd/dhclient-enter-hook.in
index a746856..c814f2b 100755
--- a/avahi-autoipd/dhclient-enter-hook.in
+++ b/avahi-autoipd/dhclient-enter-hook.in
@@ -22,7 +22,7 @@ case "$reason" in
         ;;
 
     PREINIT|BOUND|RENEW|REBIND|REBOOT|STOP|RELEASE)
-        @sbindir@/avahi-autoipd -k $interface 2> /dev/null
+        [ ! -x @sbindir@/avahi-autoipd ] || @sbindir@/avahi-autoipd -k $interface 2> /dev/null
         ;;
     
     EXPIRE|FAIL|TIMEOUT)
diff --git a/avahi-autoipd/dhclient-exit-hook.in b/avahi-autoipd/dhclient-exit-hook.in
index 379cb46..673bda3 100755
--- a/avahi-autoipd/dhclient-exit-hook.in
+++ b/avahi-autoipd/dhclient-exit-hook.in
@@ -26,6 +26,6 @@ case "$reason" in
         ;;
 
     EXPIRE|FAIL|TIMEOUT)
-        @sbindir@/avahi-autoipd -wD $interface 2> /dev/null
+        [ ! -x @sbindir@/avahi-autoipd ] || @sbindir@/avahi-autoipd -wD $interface 2> /dev/null
         ;;
 esac
-- 
2.8.1

